src/libraries/System.Text.Json/tests/TrimmingTests/SerializerEntryPoint/DeserializeAsync.FromStream.TypedObject.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace SerializerTrimmingTest
+{
+    /// <summary>
+    /// Tests that the serializer's JsonSerializer.DeserializeAsync<T>(Stream utf8Json, JsonSerializerOptions options, CancellationToken cancellationToken)
+    /// overload has the appropriate linker annotations.
+    /// A collection and a POCO are used. Public constructors and public properties are expected to be preserved.
+    /// </summary>
+    internal class Program
+    {
+        static int Main(string[] args)
+        {
+            string json = "[1]";
+
+            int[] arr = (int[])Task.Run(async () =>
+            {
+                using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(json)))
+                {
+                    return await JsonSerializer.DeserializeAsync<int[]>(stream);
+                }
+            }).GetAwaiter().GetResult();
```

Comment:
Can you use a `static async Task<int>Main` method instead? Then you can just await as normal without having to do a Task.Run.